### ISSUE TITLE
[Mac OS] Organic Maps does not open KMZ files

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#7465]
And is not offered in the Open With... menu.

KML files are opened, and OM is listed in the Open With... menu.

**Comment by anton-p14** [comment]
Hi @biodranik  I tested this on my machine and I can’t reproduce it on the latest version.

**System**
- macOS **Sequoia 15.5 (24F74)**
- Organic Maps **2026.01.16-7**

On my system, `.kmz` files are associated with Organic Maps (double-click opens OM), and importing a minimal KMZ containing a valid `doc.kml` works successfully (“Bookmarks loaded successfully”).

Screenshot:

<img src="https://github.com/user-attachments/assets/7f9c5362-7aa5-4b74-8077-da323fd58295" width="700" />

So it looks like this might already be fixed (at least in recent builds). Please let me know if there is a specific KMZ sample that still fails to open.

**Comment by biodranik** [comment]
1. Is it also shown in the Finder's "Open With..." right-mouse click menu?
2. Does it work if Google Earth is also installed?

**Comment by anton-p14** [comment]
Hi @biodranik,

1) Finder “Open With…”
Yes — Organic Maps is shown in Finder’s right-click Open With… menu for .kmz, and it is set as the default app on my system:
<img width="900" alt="Finder Open With shows Organic Maps for KMZ" src="https://github.com/user-attachments/assets/52188133-60b1-4caa-9a16-34d4033fd1ca" />

2) Google Earth installed
I don’t currently have Google Earth Pro Desktop installed on this Mac, so I can’t test. In the Mac App Store, Google Earth appears as an iOS/iPadOS app only (screenshot):
<img width="900" alt="Google Earth in App Store shows iOS/iPadOS only" src="https://github.com/user-attachments/assets/6562d936-c858-4d6e-a96c-a0b85ca4fcbe" />

**Comment by biodranik** [comment]
It could be that desktop Google Earth on Mac overrides Organic Maps…

**Comment by anton-p14** [comment]
Hi @biodranik,

1) Finder “Open With…”
With Google Earth Pro installed, .kmz files are now associated with Google Earth Pro (default), and Finder’s right-click Open With… menu shows Google Earth Pro.

2) Organic Maps missing / can’t be set as default for KMZ
In this setup, Organic Maps is not listed under Open With… for .kmz.
When I choose Other… and manually select Organic Maps, macOS shows the warning:
     “It’s not known if this application can open ‘test.kmz’.”

I also tried setting Organic Maps via Get Info → Open with → Organic Maps → Change All… and enabling Always open with, but the file still doesn’t open. Instead macOS blocks it with:
     “Apple could not verify ‘test.kmz’ is free of malware that may harm your Mac or compromise your privacy.”

So it looks like after installing Google Earth Pro, .kmz handling/association changes and Organic Maps may not be properly registered as a handler for .kmz in Finder, leading to macOS refusing to open the file.

**Comment by anton-p14** [comment]
Would adding .kmz as a supported file type in the macOS app’s Info.plist (so Finder registers Organic Maps as a KMZ handler) be the right way to fix this globally?

**Comment by biodranik** [comment]
Isn't it there yet? MacOS silicone app == iPad app == iOS app now.

**Comment by anton-p14** [comment]
Not exactly. Apple Silicon Macs can run some iOS/iPad apps, but that doesn’t mean every iOS app is available on macOS.

In this case, the Google Earth shown in the App Store is the iOS/iPadOS app, while the one that affects .kmz file association in Finder is Google Earth Pro Desktop for macOS, which is still Intel-only and runs via Rosetta 2 on Apple Silicon.
